### ISSUE TITLE
Progress?

### ISSUE THREAD

**Issue by joe71101** [Tempest0580/Temptv#73]
So is this addon officially dead in the water?  Most live TV doesn't work, many times the movies/TV shows are pulling dead links.  If it is dead then that's OK, just wondering if we should hold out hope for a return to glory or not?

**Comment by jewbmx** [comment]
I haven't "seen" or heard from Tempest in a long while so ya might wanna use something else for a while like the oath or something and then just wait and see if Tempest does return lol.
If ya need some iptv to use you can probably add this in pretty easily. Its what ive been using altho its a bit glitchy like most iptv stuff lol
https://github.com/jewbmx/ScraperWork/blob/main/indexers/webOasis.py

**Comment by Tredap** [comment]
Hey everyone if you're looking for a good USTVgo alternative to Tempest try Arxtic-Ape. It's working fine for me thus far. https://github.com/arxtic-ape

**Comment by simakuutio** [comment]
Yep, that works like a charm, it have different approach which bypass very nicely that "vpn" check.

**Comment by simakuutio** [comment]
And now that one broke too because ustvgo changed something... Let's hope developer is still around, I did evaluate change a little bit and it shouldn't be too hard to fix (some .php things changed to .js).

**Comment by jewbmx** [comment]
If you guys need a simple iptv source you can just grab my webOasis indexer thats in my ScraperWork repo, then toss it in whatever addon ya got with some easy lines of code 😉. The radio and tv features work pretty decent.

**Comment by simakuutio** [comment]
Ok, let's pinpoint to the code where change is needed... old code looks like this (part of the function):

       def resolve(self,url):
                #url = 'https://ustvgo.tv/{}'.format(url)
                html = requests.get(url, headers={'user-agent': constants.USER_AGENT}).text
                fid = re.findall('/clappr\.php\?stream=([^\"\']+)', html)

ustvgo have now that clappr replaced with .js version. Someone with more skills than I have should see how those parameters should be passed to it, I bet it's rather close but not exactly the same. Looking forwards to hear some ideas.

**Comment by jewbmx** [comment]
I see this from my site snoop that seems right but completely untested lol..



view-source:https://ustvgo.tv/amc-live/

<iframe src='/player.php?stream=AMC' allowfullscreen="" scrolling="no" frameborder="0"></iframe>


view-source:https://ustvgo.tv/player.php?stream=AMC

var hls_src='https://h1.ustvgo.la/AMC/myStream/playlist.m3u8?wmsAuthSign=c2VydmVyX3RpbWU9MS8xNS8yMDIyIDM6MzQ6NTIgUE0maGFzaF92YWx1ZT1xNFpNUWx2LzhYSWZYUnNvY09LRmFRPT0mdmFsaWRtaW51dGVzPTI0MA==';

Thats on mobile and pretty easy to test with some channels to see if it works lol.

Might wanna check reddit or a more active iptv type of github page also lol

**Comment by jewbmx** [comment]
If that bullshit at the beginning of re.findall is what you see as an issue, try removing it some to be like  re.findall('stream=(blahblah)', html)

**Comment by simakuutio** [comment]
Well, that's not the only place where it's needed... let's put whole function here so you can see it easier:

        def resolve(self,url):
                #url = 'https://ustvgo.tv/{}'.format(url)
                html = requests.get(url, headers={'user-agent': constants.USER_AGENT}).text
                fid = re.findall('/clappr\.php\?stream=([^\"\']+)', html)
                if len(fid) == 0:
                        return ' '
                fid = fid[0]
                post_url = 'https://ustvgo.tv/data.php'
                data = {'stream': fid}
                play_url = requests.post(post_url, data=data, headers={'origin':'https://ustvgo.tv', 'referer':'https://ustvgo.tv/clappr.php?stream={}'.format(fid), 'x-requested-with': 'XMLHttpRequest'}).text
                head = {'referer': 'https://ustvgo.tv/clappr.php?stream={}', 'user-agent': constants.USER_AGENT}
                return '{}|{}'.format(play_url, urlencode(head)), False

So it's needed also for that header ... "removing bullshit" doesn't solve anything here...
